firedrake/mesh.py

```diff
@@ -1393,19 +1394,19 @@ def Mesh(meshfile, **kwargs):
         geometric_dim = tcell.topological_dimension()
     cell = tcell.reconstruct(geometric_dimension=geometric_dim)
 
-    element = ufl.VectorElement("Lagrange", cell, 1)
+    element = ufl.FiniteElement("Lagrange", cell, 1, variant="equispaced")
+    vector_element = ufl.VectorElement(element, dim=geometric_dim)
     # Create mesh object
-    mesh = MeshGeometry.__new__(MeshGeometry, element)
+    mesh = MeshGeometry.__new__(MeshGeometry, vector_element)
     mesh._topology = topology
 
-    def callback(self):
+    def callback(self, vector_element=None, geometric_dim=None):
         """Finish initialisation."""
         del self._callback
         # Finish the initialisation of mesh topology
         self.topology.init()
 
-        coordinates_fs = functionspace.VectorFunctionSpace(self.topology, "Lagrange", 1,
-                                                           dim=geometric_dim)
+        coordinates_fs = functionspace.FunctionSpace(self.topology, vector_element)
```

Comment:
Doesn't the `coordinates_fs` in `ExtrudedMesh` need to be similarly updated? https://github.com/firedrakeproject/firedrake/blob/c611797177a5a06caf4fc2d41646ce09d49deab5/firedrake/mesh.py#L1520

Reply:
Also does this complicate the link to the `._ufl_mesh` and `ufl_cell` properties (see #1667) ?
https://github.com/firedrakeproject/firedrake/blob/c611797177a5a06caf4fc2d41646ce09d49deab5/firedrake/mesh.py#L449

Reply:
Also, and this might be irrelevant, there's some of the funky utility meshes work by creating a coordinates field then make the mesh from that.

Reply:
Thanks. Yes, `ExtrudedMesh` needs similar update. As here we're requiring that the coordinates should always be in equispaced fs this should affect `._ufl_mesh` as well. Same for the custom meshes.

Reply:
Handling all of this would be easier if we could pass `variant="equispaced"` directly to `FunctionSpace`. Maybe worth changing the API?

Reply:
@wence- might have some sage advice